### PR TITLE
Update CHANGELOG.md for v5.9.0

### PR DESCRIPTION
**Pull request by SarahFrench**
Notes:

- I've removed the changelog entry for this PR which was reverted on the Thursday : https://github.com/GoogleCloudPlatform/magic-modules/pull/9578
    - I've picked the revert of that PR into the TPG release-5.9.0 branch ([see here](https://github.com/hashicorp/terraform-provider-google/commits/release-5.9.0))

### CODE REVIEW

**Comment by SarahFrench** [comment]
I've run the test affected by the reverted change here in TeamCity: https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_Google_NightlyTests_GOOGLE_PACKAGE_SQL/70021?buildTab=overview&hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,32 @@
...
+* **New Data Source:** `google_logging_organization_settings` ([#16658](https://github.com/hashicorp/terraform-provider-google/pull/16658))
+* **New Data Source:** `google_logging_project_settings` ([#16658](https://github.com/hashicorp/terraform-provider-google/pull/16658))
+* **New Data Source:** `google_vmwareengine_network_policy` ([#16639](https://github.com/hashicorp/terraform-provider-google/pull/16639))
+* **New Data Source:** `google_vmwareengine_nsx_credentials` ([#16669](https://github.com/hashicorp/terraform-provider-google/pull/16669))
+* **New Resource:** `google_scc_event_threat_detection_custom_module` ([#16649](https://github.com/hashicorp/terraform-provider-google/pull/16649))
+* **New Resource:** `google_secure_source_manager_instance` ([#16637](https://github.com/hashicorp/terraform-provider-google/pull/16637))
+* **New Resource:** `google_vmwareengine_network_policy` ([#16639](https://github.com/hashicorp/terraform-provider-google/pull/16639))
+
+IMPROVEMENTS:
+* bigqueryconnection - add `spark` support to `google_bigquery_connection` resource ([#16677](https://github.com/hashicorp/terraform-provider-google/pull/16677))
```
Comment (shuyama1) [inline CHANGELOG.md:14]:
```suggestion
* bigqueryconnection - added `spark` support to `google_bigquery_connection` resource ([#16677](https://github.com/hashicorp/terraform-provider-google/pull/16677))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,32 @@
...
+* cloudidentity: added `expiry_detail` field to `google_cloud_identity_group_membership` resource ([#16643](https://github.com/hashicorp/terraform-provider-google/pull/16643))
+* container: added `queued_provisioning` field to `google_container_node_pool` resource ([#16623](https://github.com/hashicorp/terraform-provider-google/pull/16623))
+* container: promoted GKE Autoscaling Profiles to GA in the `cluster_autoscaling` block in `google_container_cluster` resource ([#16653](https://github.com/hashicorp/terraform-provider-google/pull/16653))
```
Comment (shuyama1) [inline CHANGELOG.md:17]:
```suggestion
* container: promoted `autoscaling_profile` to GA in the `cluster_autoscaling` block in `google_container_cluster` resource ([#16653](https://github.com/hashicorp/terraform-provider-google/pull/16653))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,32 @@
...
+* dlp: update required fields in `timespan_config` config block for `data_loss_prevention_job_trigger` resource ([#16628](https://github.com/hashicorp/terraform-provider-google/pull/16628))
```
Comment (shuyama1) [inline CHANGELOG.md:18]:
```suggestion
* dlp: updated required fields in `timespan_config` config block for `data_loss_prevention_job_trigger` resource ([#16628](https://github.com/hashicorp/terraform-provider-google/pull/16628))
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,32 @@
...
+* **New Data Source:** `google_logging_organization_settings` ([#16658](https://github.com/hashicorp/terraform-provider-google/pull/16658))
+* **New Data Source:** `google_logging_project_settings` ([#16658](https://github.com/hashicorp/terraform-provider-google/pull/16658))
+* **New Data Source:** `google_vmwareengine_network_policy` ([#16639](https://github.com/hashicorp/terraform-provider-google/pull/16639))
+* **New Data Source:** `google_vmwareengine_nsx_credentials` ([#16669](https://github.com/hashicorp/terraform-provider-google/pull/16669))
+* **New Resource:** `google_scc_event_threat_detection_custom_module` ([#16649](https://github.com/hashicorp/terraform-provider-google/pull/16649))
+* **New Resource:** `google_secure_source_manager_instance` ([#16637](https://github.com/hashicorp/terraform-provider-google/pull/16637))
+* **New Resource:** `google_vmwareengine_network_policy` ([#16639](https://github.com/hashicorp/terraform-provider-google/pull/16639))
+
+IMPROVEMENTS:
+* bigqueryconnection - added `spark` support to `google_bigquery_connection` resource ([#16677](https://github.com/hashicorp/terraform-provider-google/pull/16677))
```
Comment (SarahFrench) [inline CHANGELOG.md:14]:
```suggestion
* bigqueryconnection: added `spark` support to `google_bigquery_connection` resource ([#16677](https://github.com/hashicorp/terraform-provider-google/pull/16677))
```

**Review by ScottSuarez — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,32 @@
...
+* gkehub: added `binary_authorization_config` field to `google_gke_hub_fleet` resource ([#16674](https://github.com/hashicorp/terraform-provider-google/pull/16674))
+* sql: added support for in-place updates to the `edition` field in `google_sql_database_instance` resource ([#16629](https://github.com/hashicorp/terraform-provider-google/pull/16629))
+
+BUG FIXES:
+* artifactregistry: fixed permadiff due to unsorted `virtual_repository_config` array in `google_artifact_registry_repository` ([#16646](https://github.com/hashicorp/terraform-provider-google/pull/16646))
+* container: made `dns_config` field updatable on `google_container_cluster` resource ([#16652](https://github.com/hashicorp/terraform-provider-google/pull/16652))
+* firestore: retried resource creation for error 409 with the text "Aborted due to cross-transaction contention" in `google_firestore_index` resource ([#16618](https://github.com/hashicorp/terraform-provider-google/pull/16618))
+* firestore: retried resource creation for error 409 with the text "Please retry, underlying data changed" in `google_firestore_index` resource ([#16670](https://github.com/hashicorp/terraform-provider-google/pull/16670))
```
[the remark this answers is not in the document]
Comment (SarahFrench) [reply CHANGELOG.md:27]: They're for different text in the error message - should I still consolidate them?

[the remark this answers is not in the document]
Comment (ScottSuarez) [reply CHANGELOG.md:27]: I think so. It's not clear to me on initial reading what the differences is. What might a user think reading these notes.

**Pull request merged**